### PR TITLE
feat: Add CI to detect breaking changes

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: .github/workflows/schema-changes.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Download and build json-schema-diff
+        if: steps.cache-json-schema-diff-bin.outputs.cache-hit != 'true'
+        run: cargo install json-schema-diff --version 0.1.1 --features build-binary
+      - name: Generate schema diff
+        uses: getsentry/action-migrations@v1.0.8
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          # i have no clue what this parameter is for, but i have to set it to
+          # a nonempty string to get any results
+          migration: "bogusvalue"
```
Comment (lynnagara) [inline .github/workflows/schema-changes.yml:36]: 😆

File: schemas/outcomes.v1.schema.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
       "type": "string"
     },
     "org_id": {
-      "type": ["integer", "null"]
+      "type": ["integer"]
```
Comment (lynnagara) [inline schemas/outcomes.v1.schema.json:10]: was this for testing?

**Comment by lynnagara** [comment]
Just to confirm, are breaking changes always considered a "failure" in CI?

**Comment by untitaker** [comment]
> Just to confirm, are breaking changes always considered a "failure" in CI?

yes, but for as long as this is experimental I would not make that job required

File: schemas/outcomes.v1.schema.json (hunk 2 above)
Comment (untitaker) [reply schemas/outcomes.v1.schema.json:10]: yup

**Review by lynnagara — APPROVED** [review]

**Pull request merged**